### ISSUE TITLE
Problems about Shears and SQFT

### ISSUE THREAD

**Issue by RicardoZhy1215** [IntelLabs/Hardware-Aware-Automated-Machine-Learning#39]
Hi, thanks for your amazing works!  I have questions about these two papers. In the Shears, you first sparsify the pretrained model and use NLS to recover the accuracy, which is very close to Dense pretrained model with Lora fine-tuning. It has very promising results. Because it means we can introduce high level sparsity in pretrained model and through NLS to recover the accuracy during fine-tuning stage.

But in the SQFT, you compare the lora vs shears vs SQFT + SparsePEFT at certain sparsity like 50%. You don't compare the Dense pretrained model with Lora fine-tuning.

<img width="741" alt="Image" src="https://github.com/user-attachments/assets/610010d9-aba3-4249-a2ee-2b1fdec48d7f" />

I am very confused that the shears only have 52.2% accuracy. Actually, I fine tuned dense llama3 8B on gsm8k by using lora. It has around 63% accuracy. In this way, shears seems can not recover accuracy. Could you answer why this happend?

In the SQFT, you only shows that for a sparse pretrained model, SQFT+SparsePEFT do has smilar performance compare to Lora. But it seems SQFT+SparsePEFT has big accuracy gap compare to dense pretrained model with lora fine-tuning.
Thanks for your time to answer my questions.

**Comment by Yuan0320** [comment]
Hi @RicardoZhy1215, thanks for your interest in our papers and your invaluable insights.

As you have observed, fine-tuning the dense model will result in higher accuracy than the weight-sparse model. The difference is that sparse models, while more efficient, often experience a drop in accuracy due to reduced capacity. For this task (GSM8K), Llama-3-8B appears to be more sensitive to high sparsity levels (50%), resulting in the drop you observe compared to the dense model. **In this case, reducing the sparsity levels might be one way to find a good trade-off between sparsity and accuracy drop.**

SQFT’s NLS and SparsePEFT are designed to help recover some of this lost accuracy, but they may not fully match the performance of fine-tuned dense models, especially at higher sparsity levels.

You can also see in Table 10 of SQFT paper that we can use NLS in a (quantized) dense model to obtain better results than LoRA.
 
Thank you for bringing this to our attention, and we appreciate your exploration of our work! 😊

**Comment by RicardoZhy1215** [comment]
Thanks for your quick response! @Yuan0320
